templates/app-ts/app.ts

```diff
@@ -1,25 +1,34 @@
-import AutoLoad from 'fastify-autoload';
 import * as path from 'path';
-import { FastifyInstance } from 'fastify';
+import AutoLoad, {AutoloadPluginOptions} from 'fastify-autoload';
+import { FastifyPluginAsync } from 'fastify';
 
-export default async function (fastify: FastifyInstance, opts: any) {
+export type AppOptions = {
+  // Place your custom options for app below here.
+} & Partial<AutoloadPluginOptions>;
+
+const app: FastifyPluginAsync<AppOptions> = async (
+    fastify,
+    opts
+): Promise<void> => {
   // Place here your custom code!
 
   // Do not touch the following lines
 
   // This loads all plugins defined in plugins
   // those should be support plugins that are reused
   // through your application
-  fastify.register(AutoLoad, {
+  await fastify.register(AutoLoad, {
```

Comment:
This await is not really needed right now.

Reply:
Not really familiar with the internal flow of this so I may misunderstand. 
As I understand it:
should it be "plugin return promise that is resolved when it is done loading/doing everything it needs" rather than
"plugin returns a promise that is resolved when everything is scheduled to be loaded"?

Reply:
It does not return a promise. The server is a thenable, and by doing an `await` all the plugins registered so far are loaded. The same happens after the plugin function resolves. The only reason somebody would add this is if they want to use a decorator _inside_ the plugin function but this is not the case.

Note that keeping them does not add any problem, however I find it confusing

Reply:
Oh, now I understand. Only if you depend on the modifications from other plugins then you would have to wait for them to load.
Changed these to `void fastify.register...` to pacify some very strict ESLint rules (no-floating-promises).